Bottom: 71c3c0cd1acb11f4388933f3f9dfd99677e32662
Top:    e8539cb6d56fbbd5db2ccd39e02a727fbddafb40
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 22:36:05 +0000

Refresh of try-to-prepend-not-append-to

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index a576c0f..f284d54 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -119,19 +119,19 @@ object3d_set_appearance (object3d *object, appearance *appear)
 void
 object3d_add_edge (object3d *object, edge_ref edge)
 {
-  object->edges = g_list_append (object->edges, (void *)edge);
+  object->edges = g_list_prepend (object->edges, (void *)edge);
 }
 
 void
 object3d_add_vertex (object3d *object, vertex3d *vertex)
 {
-  object->vertices = g_list_append (object->vertices, vertex);
+  object->vertices = g_list_prepend (object->vertices, vertex);
 }
 
 void
 object3d_add_face (object3d *object, face3d *face)
 {
-  object->faces = g_list_append (object->faces, face);
+  object->faces = g_list_prepend (object->faces, face);
 }
 
 
@@ -696,7 +696,7 @@ object3d_from_contours (POLYAREA *contours,
           }
 #endif
 
-      objects = g_list_append (objects, object);
+      objects = g_list_prepend (objects, object);
 
       link = malloc (sizeof (polygon_3d_link));
       pa->user_data = link;
